### PR TITLE
[RFR] Remove entry /etc/hosts when container stops

### PR DESCRIPTION
**Pull request by Jean-Beru**
Issue #4

Replace PR #8

### CODE REVIEW

File: src/Docker/Docker.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+class Docker extends DockerBase
+{
+    /**
+     * @param string   $entrypoint
+     * @param callable $callback
+     */
+    public function listenEvents($entrypoint, callable $callback)
+    {
+        $stream = stream_socket_client($entrypoint, $errno, $errstr, 30);
```
Comment (iamluc) [inline src/Docker/Docker.php]: Did you try to use `DockerAdapter::send()` instead of creating your own stream ?

Comment (Jean-Beru) [inline src/Docker/Docker.php]: Yes... But when I want to read the stream with `read()`, I have no result. With `read(1024)`, I have no result until stream has 1024 caracters.
With `read(1)` in a loop, it's working... but very dirty!

Comment (Jean-Beru) [inline src/Docker/Docker.php]: It's OK now with `\GuzzleHttp\Stream\read_line`.

File: src/Application.php [hunk 2]
```diff
@@ -2,108 +2,196 @@
...
+     * @param Container $container
+     *
+     * @return bool
+     */
+    private function isExposed(Container $container)
+    {
+        $inspection = $container->getRuntimeInformations();
+        if (
+            empty($inspection['NetworkSettings']['Ports']) ||
+            empty($inspection['NetworkSettings']['IPAddress']) ||
```
Comment (iamluc) [inline src/Application.php]: This test is wrong, as the container could have an IPAddress only in a network

**Comment by iamluc** [comment]
Thanks @Jean-Beru !
Except one comment, LGTM.

File: src/Application.php (hunk 2 above)
Comment (Jean-Beru) [inline src/Application.php]: OKay, I remove it

**Pull request merged**